### PR TITLE
add server address and port span attributes to redis otel trace instrumentation

### PR DESCRIPTION
**Pull request by esara**
address https://github.com/redis/go-redis/issues/2825

### CODE REVIEW

File: extra/redisotel/tracing.go [hunk 1]
```diff
@@ -41,6 +68,19 @@ func InstrumentTracing(rdb redis.UniversalClient, opts ...TracingOption) error {
...
+				opts = append(opts, WithAttributes(
+					semconv.ServerAddress(host),
+				))
+				// Parse the port string to an integer
+				port, err := strconv.Atoi(portString)
+				if err == nil {
+					opts = append(opts, WithAttributes(
+						semconv.ServerPort(port),
+					))
+				}
```
Comment (vmihailenco) [inline extra/redisotel/tracing.go:82]: Could you please add a helper function so we don't repeat this block 3 times?

Comment (esara) [reply extra/redisotel/tracing.go]: done :-) thanks!

File: extra/redisotel/config.go [hunk 2]
```diff
@@ -63,17 +62,11 @@ func newConfig(opts ...baseOption) *config {
...
+	conf.attrs = append(conf.attrs, semconv.DBSystemRedis)
 
 	return conf
 }
 
-func WithDBSystem(dbSystem string) Option {
-	return option(func(conf *config) {
-		conf.dbSystem = dbSystem
-	})
-}
```
Comment (vmihailenco) [inline extra/redisotel/config.go:75]: @esara could you please restore this? It is a public API and it is still useful for those who use this client with other systems that provide Redis compatible API.

**Comment by vmihailenco** [comment]
@esara LGTM, but I think we should keep `WithDBSystem`...

File: extra/redisotel/config.go (hunk 2 above)
Comment (esara) [reply extra/redisotel/config.go]: of course, sorry for not realizing, restored

File: extra/redisotel/config.go [hunk 3]
```diff
@@ -63,7 +62,7 @@ func newConfig(opts ...baseOption) *config {
 		opt.apply(conf)
 	}
 
-	conf.attrs = append(conf.attrs, semconv.DBSystemKey.String(conf.dbSystem))
+	conf.attrs = append(conf.attrs, semconv.DBSystemRedis)
```
Comment (vmihailenco) [inline extra/redisotel/config.go:65]: @esara looks like this should be reverted as well.

**Review by vmihailenco — APPROVED** [review]

File: extra/redisotel/config.go (hunk 3 above)
Comment (esara) [reply extra/redisotel/config.go]: no problem, I was trying for semconv, where possible, but it is not crucial to this pr - reverted

**Review by vmihailenco — APPROVED** [review]

**Pull request merged**